commit 025af189fb44250206dd8a32fa4a682392af3301
Author: Thomas Hellstrom <thellstrom@vmware.com>
Date:   Fri Nov 20 11:43:50 2015 -0800

    drm/ttm: Fixed a read/write lock imbalance
    
    In ttm_write_lock(), the uninterruptible path should call
    __ttm_write_lock() not __ttm_read_lock().  This fixes a vmwgfx hang
    on F23 start up.
    
    syeh: Extracted this from one of Thomas' internal patches.
    
    Cc: <stable@vger.kernel.org>
    Signed-off-by: Thomas Hellstrom <thellstrom@vmware.com>
    Reviewed-by: Sinclair Yeh <syeh@vmware.com>

diff --git a/drivers/gpu/drm/ttm/ttm_lock.c b/drivers/gpu/drm/ttm/ttm_lock.c
index 6a95454..f154fb1 100644
--- a/drivers/gpu/drm/ttm/ttm_lock.c
+++ b/drivers/gpu/drm/ttm/ttm_lock.c
@@ -180,7 +180,7 @@ int ttm_write_lock(struct ttm_lock *lock, bool interruptible)
 			spin_unlock(&lock->lock);
 		}
 	} else
-		wait_event(lock->queue, __ttm_read_lock(lock));
+		wait_event(lock->queue, __ttm_write_lock(lock));
 
 	return ret;
 }
